core/api/transport/session.cpp

```diff
@@ -51,24 +53,27 @@ namespace kagome::server {
   void Session::processResponse(std::string response) {
     // 10 seconds wait for new request, then close session
     heartbeat_.expires_after(std::chrono::seconds(10));
-    do_write(response);
+    do_write(std::move(response));
   }
 
-  void Session::do_write(const std::string &response) {
-    boost::asio::async_write(
-        socket_, boost::asio::buffer(response.data(), response.size()),
-        [this, self = shared_from_this()](
-            boost::system::error_code ec, std::size_t
-            /*length*/) {
-          if (!ec) {
-            do_read();
-          } else {
-            stop();
-          }
-        });
+  void Session::do_write(std::string response) {
+    auto r = std::make_shared<std::string>(std::move(response));
+
+    boost::asio::async_write(socket_,
+                             boost::asio::const_buffer(r->data(), r->size()),
+                             [this, self = shared_from_this(), r](
```

Comment:
you pass both this and self, but use this. should be self. e.g. self->do_read(), etc.

Reply:
ok